helpers/responseQueue.js

```diff
@@ -0,0 +1,34 @@
+const queue = [];
+let queueIsProcessing = false;
+
+function addItemToQueue(item) {
+	queue.push(item);
+
+	if(!queueIsProcessing) {
+		processQueue(0);
+	}
+}
+
+function removeItemFromQueue(id = 0) {
+	queue.splice(id,1);
+	setQueueStatus(false);
+}
+
+async function processQueue(queueId = 0) {
+	if(queue.length > 0 && !queueIsProcessing) {
+		let currentItem = queue[queueId];
+		setQueueStatus();
+		const exec = await currentItem.exec(currentItem.params, currentItem.finalFilepath, currentItem.hash, currentItem.res);
```

Comment:
I *think* normal best practise is not to overload standard method names as field names, 
so something like currentItem.responseFn would be clearer/safer.